server/src/main/java/org/elasticsearch/action/admin/indices/create/CreateIndexRequest.java

```diff
@@ -291,7 +283,7 @@ else if (MapperService.SINGLE_MAPPING_NAME.equals(type) == false) {
      * ("field1", "type=string,store=true").
      */
     public CreateIndexRequest mapping(String type, Object... source) {
-        mapping(type, PutMappingRequest.buildFromSimplifiedDef(type, source));
+        mapping(PutMappingRequest.buildFromSimplifiedDef(MapperService.SINGLE_MAPPING_NAME, source));
```

Comment:
For context, what's the plan to remove these remaining methods that accept `Object... source`?

Reply:
My current plan is to have a new method, `mappingFromSimplifiedDef(Objects... source)`, and remove the existing method, to make sure that we get compilation failures rather than silently treating the `type` parameter as part of the source.